### PR TITLE
Docs for channel upgrades CLI

### PR DESCRIPTION
**Pull request by chatton**
## Description

ref #5548

Also realizing, after this gets merged I can update #5504 to reference this CLI instead of gov submit proposal.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: docs/docs/01-ibc/06-channel-upgrades.md [hunk 1]
```diff
@@ -164,3 +164,39 @@ IBC application callbacks should be primarily used to validate data fields and d
...
+  --title "Channel Upgrades Governance Proposal" \
+  --summary "Upgrade all transfer channels to be fee enabled" \
+  --port-pattern "transfer"
+```
+
+It is also possible to explicitly list a comma separated string of channel IDs. It is important to note that the 
+regular expression matching specified by `--port-pattern` (which defaults to `transfer`) still applies.
+
+For example the following command would generate the contents of a `proposal.json` file which would attempt to upgrade
+channels with a port ID of `transfer` and channelID of `channel-0`, `channel-1` or `channel-2`.
```
Comment (DimitrisJim) [inline docs/docs/01-ibc/06-channel-upgrades.md:192]:
```suggestion
channels with a port ID of `transfer` and a channelID of `channel-0`, `channel-1` or `channel-2`.
```

File: docs/docs/01-ibc/06-channel-upgrades.md [hunk 2]
```diff
@@ -164,3 +164,39 @@ IBC application callbacks should be primarily used to validate data fields and d
 > IBC applications should not attempt to process any packet data under the new conditions until after `OnChanUpgradeOpen`
 > has been executed, as up until this point it is still possible for the upgrade handshake to fail and for the channel
 > to remain in the pre-upgraded state. 
+
+## Upgrading channels with the CLI
+
+A new cli has been added which enables either
+    - submitting a governance proposal which contains a `MsgChanUpgradeInit` for every channel to be upgraded.
+    - generation a `proposal.json` file which contains the proposal contents to be edited/submitted at a later date.
```
Comment (DimitrisJim) [inline docs/docs/01-ibc/06-channel-upgrades.md:172]:
```suggestion
    - generating a `proposal.json` file which contains the proposal contents to be edited/submitted at a later date.
```

File: docs/docs/01-ibc/06-channel-upgrades.md [hunk 3]
```diff
@@ -164,3 +164,39 @@ IBC application callbacks should be primarily used to validate data fields and d
...
+
+The following example, would submit a governance proposal with the specified deposit, title and summary which would
+contain a `MsgChanUpgradeInit` for all `OPEN` channels whose port matches the regular expression `transfer`.
```
Comment (DimitrisJim) [inline docs/docs/01-ibc/06-channel-upgrades.md:175]:
```suggestion
The following example, would submit a governance proposal with the specified deposit, title and summary while also containing a `MsgChannelUpgradeInit` for all `OPEN` channels whose port matches the regular expression `transfer`.
```

**Review by DimitrisJim — APPROVED** [review]
lgtm, left minor nits

File: docs/docs/01-ibc/06-channel-upgrades.md (hunk 3 above)
Comment (charleenfei) [reply docs/docs/01-ibc/06-channel-upgrades.md:175]: i kind of prefer @chatton's version tbh bc the proposal is there to contain (approve) the `MsgChannelUpgradeInit`, but it's a super nit really.

**Review by charleenfei — APPROVED** [review]
great work @chatton! thanks for all these docs

File: docs/docs/01-ibc/06-channel-upgrades.md (hunk 3 above)
Comment (chatton) [reply docs/docs/01-ibc/06-channel-upgrades.md:175]: will leave as is if you guys are okay with it then?

Comment (DimitrisJim) [reply docs/docs/01-ibc/06-channel-upgrades.md:175]: ah, maybe just the `MsgChannelUpgradeInit` as opposed to `MsgChanUpgradeInit`? (git diff isn't particularly good at showing the specific diffs).

Wording is perfectly fine by me either way!

File: docs/docs/01-ibc/06-channel-upgrades.md [hunk 4]
```diff
@@ -164,3 +164,39 @@ IBC application callbacks should be primarily used to validate data fields and d
 > IBC applications should not attempt to process any packet data under the new conditions until after `OnChanUpgradeOpen`
 > has been executed, as up until this point it is still possible for the upgrade handshake to fail and for the channel
 > to remain in the pre-upgraded state. 
+
+## Upgrading channels with the CLI
+
+A new cli has been added which enables either
+    - submitting a governance proposal which contains a `MsgChanUpgradeInit` for every channel to be upgraded.
```
Comment (damiannolan) [inline docs/docs/01-ibc/06-channel-upgrades.md:171]:
```suggestion
    - submitting a governance proposal which contains a `MsgChannelUpgradeInit` for every channel to be upgraded.
```

File: docs/docs/01-ibc/06-channel-upgrades.md [hunk 5]
```diff
@@ -164,3 +164,39 @@ IBC application callbacks should be primarily used to validate data fields and d
...
+    - generating a `proposal.json` file which contains the proposal contents to be edited/submitted at a later date.
+
+The following example, would submit a governance proposal with the specified deposit, title and summary which would
+contain a `MsgChanUpgradeInit` for all `OPEN` channels whose port matches the regular expression `transfer`.
```
Comment (damiannolan) [inline docs/docs/01-ibc/06-channel-upgrades.md:175]:
```suggestion
contain a `MsgChannelUpgradeInit` for all `OPEN` channels whose port matches the regular expression `transfer`.
```

**Review by damiannolan — APPROVED** [review]
🚀

File: docs/docs/01-ibc/06-channel-upgrades.md (hunk 3 above)
Comment (chatton) [reply docs/docs/01-ibc/06-channel-upgrades.md]: ah missed the typo, thanks @DimitrisJim

**Pull request merged**